### ISSUE TITLE
no-invalid-regexp: duplicate group names are valid when defined over a union

### ISSUE THREAD

**Issue by mfulton26** [denoland/deno_lint#1372]
### Code Snippet

```ts
/(?<op>do)\(\)|(?<op>don't)\(\)|(?<op>mul)\((?<a>\d+),(?<b>\d+)\)/g;
```

### Expected Result

No lint warning.

### Actual Result

> Invalid RegExp literal
> Rework regular expression to be a valid

### Additional Info

MDN also calls this out as valid in their [valid cases](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Regex_duplicate_capture_group_name#valid_cases):

> ```js example-good
> /(?<firstName>\w+) (?<lastName>\w+)/;
> /(?<year>\d{4})-\d{2}|\d{2}-(?<year>\d{4})/;
> ```

### Version

deno 2.1.2 (stable, release, aarch64-apple-darwin)
v8 13.0.245.12-rusty
typescript 5.6.2
